soccer/planning/PlannerNode.cpp

```diff
@@ -1,99 +1,132 @@
+
+    if (context_->game_state.state == GameState::Halt) {
+        context_->trajectories.fill(Trajectory());
+        return;
+    }
 
-        const auto angle_function =
-            angleFunctionForCommandType(r->rotationCommand());
+    std::array<Trajectory*, Num_Shells> planned{};
 
-        context_->trajectories[shell] = Trajectory::Trajectory{
-            std::move(path), angle_function, granularity};
+    // Sort the robots by priority
+    std::array<int, Num_Shells> shells{};
+    for (int i = 0; i < Num_Shells; i++) {
+        shells[i] = i;
     }
+    std::sort(shells.begin(), shells.end(), [&](int a, int b) {
+        return robot_intents.at(a).priority > robot_intents.at(b).priority;
+    });
+
+    for (int i = 0; i < Num_Shells; i++) {
+        unsigned int shell = shells.at(i);
+        const auto& robot = world_state.our_robots.at(shell);
+        const auto& intent = robot_intents.at(shell);
+
+        if (!robot.visible) {
+            // For invalid robots, early return with an empty path.
+            trajectories->at(shell) = Trajectory();
+            planned.at(shell) = &trajectories->at(shell);
+            continue;
+        }
 
-    // Visualize obstacles
-    for (const auto& shape : global_obstacles.global_obstacles.shapes()) {
-        context_->debug_drawer.drawShape(shape, Qt::black, "Global Obstacles");
+        RobotInstant start{robot.pose, robot.velocity, robot.timestamp};
+
+        // TODO(Kyle): Put motion constraints in intent.
+        PlanRequest request{start,
+                            intent.motion_command,
+                            RobotConstraints(),
+                            global_obstacles,
+                            intent.local_obstacles,
+                            planned,
+                            shell,
+                            &world_state,
+                            intent.priority,
+                            debug_drawer};
+
+        Trajectory trajectory =
+            robots_planners_.at(shell).PlanForRobot(request);
+        trajectory.draw(&context_->debug_drawer);
+        trajectories->at(shell) = std::move(trajectory);
+
+        planned.at(shell) = &trajectories->at(shell);
     }
 }
 
-GlobalObstacles PlannerNode::getGlobalObstacles() {
-    const Geometry2d::ShapeSet global_obstacles = context_->globalObstacles;
-    const Geometry2d::ShapeSet goal_zone_obstacles =
-        context_->goalZoneObstacles;
-    Geometry2d::ShapeSet global_obstacles_with_goal_zones = global_obstacles;
-    global_obstacles_with_goal_zones.add(goal_zone_obstacles);
+PlannerForRobot::PlannerForRobot() : planner_idx_(-1) {
+    planners_.push_back(std::make_unique<PathTargetPlanner>());
+    planners_.push_back(std::make_unique<SettlePlanner>());
+    planners_.push_back(std::make_unique<CollectPlanner>());
+    planners_.push_back(std::make_unique<LineKickPlanner>());
+    planners_.push_back(std::make_unique<PivotPathPlanner>());
```

Comment:
Does this order matter in this section? Or is it just the empty planner must be last?

Reply:
Yeah. It only matters when multiple planners could be used to plan for the same command type. Right now, the only place that happens is with the empty/escape obstacles planner.